src/jobs.ts

```diff
@@ -0,0 +1,21 @@
+import { ConfigurationFromFile } from './config/input';
+import { Config, gatherChecks, Job } from './config/app';
+import { Tool } from './tools';
+
+export interface JobCreator {
+    createJobs(config: ConfigurationFromFile, appConfig: Config): [Job, ...Job[]];
```

Comment:
`[Job, ...Job[]]` vs `Job[]`, I'm sure you have a good reason for this and I'll feel very silly once I find out what it is and will probably start using it going forward.

Reply:
non-empty-list